### DISCUSSION TITLE
Search bar usage with mkdocs

### DISCUSSION THREAD

**Discussion by nhan000** [ObsidianPublisher/obsidian-github-publisher#99] [Q&A — Template] [answered]
### Describe the bug

On [Obsidian Mkdocs Publisher](https://obsidianmkdocs.github.io/obsidian_mkdocs_publisher_docs/), search can't look for note title or content. Only search for Headings works.

### How to reproduce ?

Use [Snippets and tools - Obsidian Mkdocs Publisher](https://obsidianmkdocs.github.io/obsidian_mkdocs_publisher_docs/Template/snippets%20and%20tools/) as an example.

Search for title of the note "Snippets and tools" doesn't return the page
![image](https://user-images.githubusercontent.com/85216095/177229698-137a7ba6-a1bf-491a-8168-86032f03d82e.png)

Search for content of the page "Automatically convert the" also doesn't return the page
![image](https://user-images.githubusercontent.com/85216095/177229747-228a3acc-c182-4bef-8b3b-6c2062705839.png)

Search for headings work
![image](https://user-images.githubusercontent.com/85216095/177229786-1c24dee2-8257-4c38-ab5c-3c0faf159332.png)

Also, what's the difference between the results that have and don't have this icon ![image](https://user-images.githubusercontent.com/85216095/177229887-5c8bd6a4-3dbb-4b5a-b683-4857c173e0a7.png) next to it?

### Configuration

N/A

### OS

Windows

**Comment by Mara-Li** [comment]
...........
This bug is not a bug ?
- The text highlight when searching
- The mkdocs search one word independently. 
- To search a complete part in the contents, you need to use `"`  

To ask for the icons and relative of the usage of the searching plugins, please use the [Material Mkdocs](https://github.com/squidfunk/mkdocs-material) discussion

**Comment by nhan000** [comment]
Thanks for your reply. I probably just don't understand how the search work then. Even after adding `"` it didn't return any results. Even on the demo site – [squidfunk.github.io/mkdocs-material](https://squidfunk.github.io/mkdocs-material/) 

<img width="860" alt="image" src="https://user-images.githubusercontent.com/85216095/177646774-98042ba8-fdd0-4327-81b4-080493d45816.png">

**Comment by Mara-Li** [comment, marked as answer]
Yeah, the search is a little strange here. Dunno how it search for title, filename and quote.
I think two things : 
- By default, material mkdocs add the filename/title frontmatter in the document’s heading 1. 
- But, the material search only search in CONTENTS, aka the other heading, maybe frontmatter title and file’s contents. 

So searching a filename (and not the heading) using quote won’t work. 

Edit : there is already an issue for this behavior on material mkdocs squidfunk/mkdocs-material#304
